File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let arr = Array(number).fill(0);
    let sum = 0;
    
    for(let i=1; i<=number; i++) {
        for(let j=i; j<=number; j=j+i) {
            arr[j-1] += 1;
        }
    }
    
    for(let i=0; i<number; i++) {
        if(limit<arr[i]) sum += power;
        else sum += arr[i];
    }
    
    return sum;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int [] arr = new int [number+1];
        int sum = 0;

        for (int i = 1; i <= number; i++) {
            for (int j = i; j <= number; j=j+i) {
                arr[j] += 1; ;
            }
        }

        for (int i = 1; i <= number; i++) {

            if( arr[i] > limit)
                sum += power;
            else
                sum+=arr[i];

        }
        
        return sum;
    }
}